internal/cmd/cmd.go

```diff
@@ -31,6 +33,19 @@ func Run(ctx context.Context, args ...string) error {
 	return cmd.ExecuteContext(ctx)
 }
 
+func authenticatedPreRun() error {
+	if err := mustBeLoggedIn(); err != nil {
+		return err
+	}
+	api, err := defaultAPIClient()
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("failed to create api client to check version")
+		return nil
+	}
+	warnVersion(context.Background(), api)
```

Comment:
I believe we'd be better to put this in the API client so it catches all calls (authenticated, unauthenticated). This check could happen once per instance of the api client (e.g. on the first call) which should almost map 1-1 to a command.

Reply:
This will result in a lot of extra logging, many CLI commands are not 1-1, for example removing a group means getting listing the groups then deleting by IDs. Being in a pre-run also means that we don't have to add this unrelated logging to the client code which is used by connectors also, and ensures it is only called once.

Reply:
I don't think we have any place to put this in the API client because we don't really use a constructor for it, but we could put it in the `apiClient` func, which is called by `defaultAPIClient`. All the CLI commands should use that function